xbmc/guilib/TextureGL.cpp

```diff
@@ -111,6 +141,35 @@ void CGLTexture::LoadToGPU()
   }
 
 #ifndef HAS_GLES
+  if (CServiceBroker::GetRenderSystem()->IsExtSupported("GL_ARB_texture_swizzle"))
```

Comment:
might also need to check for `GL_EXT_texture_swizzle`

Reply:
I've made it a bit simpler.